### PR TITLE
fix: release listener cache entries without breaking removal

### PR DESCRIPTION
**Pull request by davidmurdoch**
Snow's module-scoped load-listener cache strongly retains every wrapped listener. A bound listener can therefore keep a detached image and its DOM tree alive after the browser registration becomes unreachable. Removing one registration also deletes the cached wrapper, preventing subsequent removals on other targets or capture settings from finding the registered listener.

This PR addresses both problems in two commits:

- Capture `WeakMap` from the pristine realm and use it for the listener cache. Preserve native handling of null, undefined, and invalid primitive listeners.
- Keep wrapper identity stable across removals. Weak keys allow collection when the listener becomes unreachable, without invalidating registrations that still use the wrapper.

Both development and production bundles are regenerated. The weak-cache approach builds on the prior work by pedronfigueiredo in the closed PR #170; this also fixes shared-registration removal.

### Validation

All 13 listener tests pass against the production bundle in Chromium 147.0.7727.49, including function/object listener semantics, native argument handling, `once`, abort signals, and the existing realm-protection coverage.

The new garbage-collection test creates 30 detached images with bound load listeners, forces collection through Chrome DevTools Protocol, and checks weak references to all 60 objects. It fails against unchanged main with all 60 retained and passes with this change with none retained. This test runs only on Chrome because it uses CDP.

Three additional regression tests cover shared handlers on multiple targets, different capture settings, and removal of a nonexistent registration followed by re-addition. All three fail with only the first commit and pass with both commits.

The full Chromium suite completes with 13 spec files passing and 4 failing. Running unchanged main (`bbbdfcc`) with the same build tools and browser produces the same nine failing tests, in HTML/srcdoc, shadow DOM, object views, and the javascript-URI edge case. No additional failures were observed. Firefox and Safari were not run locally.

### CODE REVIEW

File: test/listeners.js [hunk 1]
```diff
@@ -90,4 +90,158 @@ describe('test listeners', async function () {
...
+            target.addEventListener('load', () => onceCalls++, {once: true});
+            target.addEventListener('load', () => signalCalls++, {signal: controller.signal});
+            target.dispatchEvent(new Event('load'));
+            controller.abort();
+            target.dispatchEvent(new Event('load'));
+            return [onceCalls, signalCalls];
+        });
+        expect(result).toEqual([1, 1]);
+    });
+
```
Comment (davidmurdoch) [inline test/listeners.js:171]: All of these tests would have passed before my changes, but my changes narrow what is accepted into the listener Map (er, now `WeakMap`). WeakMap doesn't allow `null`, `undefined`, numbers, and strings to be used as keys.

Previously, Snow wrapped every "handler" arg and stored it in the Map. By changing it to `WeakMap`, without filtering out invalid values, we would have made `addEventListener('load', null)` throw when Snow tried to store `null` in the `WeakMap`, even though the browser would have just ignored it.

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by naugtur** [comment]
The change looks fine overall:
- `Object(listener) === listener` is immune to malicious values that could force a skip and provide a function later.
- The test ensuring garbage collection in chrome is sufficient (garbage collection behavior of weakmap differing between browsers would be a bug in the browsers)

I'd need to update the browser drivers etc. to get the tests passing. We need at least FF and Chromium to work. 
Last merged PR was an update to the test setup and all test passed then, codebase didn't change, so I'd rather take a quick look at the tests reported as failing.

**Comment by naugtur** [comment]
Got CSP tests mostly passing. Some FF issues left, but ran out of time
https://github.com/LavaMoat/snow/pull/173

@davidmurdoch could you try to rebase and see if chrome passes?

the fixes to get FF passing are currently slopp-drafted here: https://github.com/LavaMoat/snow/pull/174
